### ISSUE TITLE
Restore keyboard navigation on chosen.js preview customizations

### ISSUE THREAD

**Issue by chillu** [silverstripe/silverstripe-framework#2101]
We take pride in the fact that the majority of our CMS UI is keyboard navigatable, and I would like to keep it this way. It not only keeps the UI accessible to users without a mouse (motion impaired), but also to screen readers who only "see" focusable elements as interactive. The preview mode chosen.js UI controls violate this at the moment.

This was already mentioned in the original [pull request](https://github.com/silverstripe/silverstripe-framework/pull/966), but gotten dropped due to the team running out of time.

Now that we're in the polish phase for 3.1, @adrexia @mateusz could one of you please look at this?
- Visible keyboard focus and tabindex on "draft"/"live" switch
- Enter key on "draft"/"live" switch switches mode
- Up/down keys trigger overlay on mode dropdown
- Up/down keys allow navigation through modes (should be built-in for chosen.js)
- Enter key on mode switches mode (should be built-in for chosen.js)

**Comment by kinglozzer** [comment]
Current state of play:

- All are keyboard navigable
- View mode toggle lacks any `:focus` styles, so it’s hard to tell when to hit the up arrow to open that menu (same issues for preview width)
- Draft/Published buttons _do_ have `:focus` styles now, so they seem to be fine to me

**Comment by chillu** [comment]
Thanks for the update, Loz! Wanna re-add those styles? I think in terms of product team attention, this might have to wait until we revisit those sections with React.

**Comment by kinglozzer** [comment]
https://github.com/silverstripe/silverstripe-admin/pull/516

**Comment by kinglozzer** [comment]
silverstripe/silverstripe-admin#516 has been merged, so I think this is fine for now
